index.bs

```diff
@@ -795,6 +741,528 @@ the following list:
+};
+</xmp>
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">url</dfn> (a [=URL
+record=]).
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">opened promise</dfn>.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">closed promise</dfn>.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">readable stream</dfn>,
+which is initially unset.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">writable stream</dfn>,
+which is initially unset.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">was ever connected
+flag</dfn>, which is initially unset.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">ready state</dfn>, which
+is a number representing the state of the connection. Initially it must be {{WebSocket/CONNECTING}}
+(0). It has the same semantics as {{WebSocket}}'s [=WebSocket/ready state=], but is not exposed to
+JavaScript.
+
+
+<dl class="domintro non-normative">
+ : <code>|socket| = new {{WebSocketStream/constructor(url, options)|WebSocketStream}}(|url| [, |options| ]</code>
+ :: Creates a new {{WebSocketStream}} object, immediately establishing the associating WebSocket
+  connection.
+
+  |url| is a string giving the <a dfn spec=url>URL</a> over which the connection is established.
+  Only "`ws`" or "`wss`" schemes are allowed; others will cause a "{{SyntaxError}}"
+  {{DOMException}}. URLs with [=fragments=] will also cause such an exception.
+
+  The |options| argument is an object whose properties can be set as follows:
+
+  : {{WebSocketStreamOptions/protocols}}
+  :: An array of strings. If it is omitted, it is equivalent to an empty array. Each string in the
+   array is a subprotocol name. The connection will only be established if the server reports that
+   it has selected one of these subprotocols. The subprotocol names have to match the requirements
+   for elements that comprise the value of \`<a http-header>`Sec-WebSocket-Protocol`</a>\` fields as
+   defined by The WebSocket protocol.  [[!WSP]]
+
+  : {{WebSocketStreamOptions/signal}}
+  :: An {{AbortSignal}} that can be used to abort the handshake. After the handshake is complete the
+   signal does nothing.
+
+ : <code>|socket| . {{WebSocketStream/url}}</code>
+ :: Returns the [=WebSocketStream/url|URL=] that was used</a> to establish the WebSocket connection.
+
+ : <code>|socket| . {{WebSocketStream/opened}}</code>
+ :: Returns a {{promise}} which resolves when the handshake successfully completes, or rejects if
+  the handshake fails. On success, it resolves to an object with the following properties:
+
+  : {{WebSocketOpenInfo/readable}}
+  :: A {{ReadableStream}} that can be used to read messages from the server. Each chunk read
+   corresponds to one message. Text messages will be read as strings; binary messages will be read
+   as ArrayBuffer objects.
```

Comment:
I forgot to change this when I changed the object type to Uint8Array. I've fixed it and added a note about why it differs from what WebSocket did. I couldn't find a way to link `{{Uint8Array/buffer}}`. If you think it would be linked let me know.